controllers/pipelines/reconciler.go

```diff
@@ -31,6 +32,21 @@ func desiredProvider(resource pipelinesv1.HasProvider, config config.Configurati
 	return config.DefaultProvider
 }
 
+func loadProvider(ctx context.Context, reader client.Reader, namespace string, desiredProvider string) (*pipelinesv1.Provider, error) {
```

Comment:
I know why this is returning a pointer but to remove the hassle of dealing with one I guess in the error case you could return the created provider, i.e.

```
func loadProvider(ctx context.Context, reader client.Reader, namespace string, desiredProvider string) (pipelinesv1.Provider, error) {
	providerNamespacedName := types.NamespacedName{
		Namespace: namespace,
		Name:      desiredProvider,
	}

	var provider = pipelinesv1.Provider{}

	err := reader.Get(ctx, providerNamespacedName, &provider)
	if err != nil {
		return provider, err
	}
	return provider, nil
}
```

Reply:
changed in [f14cca8](https://github.com/sky-uk/kfp-operator/pull/354/commits/f14cca83d6fe1a7cb015482b23effbdc540042ee)